docs/pages/updates/new-graphql-api.mdx

```diff
@@ -0,0 +1,403 @@
+  isComplete: Boolean
+  assignedTo: PersonRelateToOneInput
+  tags: TagRelateToManyInput
+  finishBy: String
+}
+
+input TasksCreateInput {
+  data: TaskCreateInput
+}
+
+input TaskCreateInput {
+  label: String
+  priority: TaskPriorityType
+  isComplete: Boolean
+  assignedTo: PersonRelateToOneInput
+  tags: TagRelateToManyInput
+  finishBy: String
+}
+
+input PersonRelateToOneInput {
+  create: PersonCreateInput
+  connect: PersonWhereUniqueInput
+  disconnect: PersonWhereUniqueInput
+  disconnectAll: Boolean
+}
+
+input TagRelateToManyInput {
+  create: [TagCreateInput]
+  connect: [TagWhereUniqueInput]
+  disconnect: [TagWhereUniqueInput]
+  disconnectAll: Boolean
+}
+
+// After
+
+input TaskUpdateArgs {
+  where: TaskWhereUniqueInput!
+  data: TaskUpdateInput!
+}
+
+input TaskUpdateInput {
+  label: String
+  priority: TaskPriorityType
+  isComplete: Boolean
+  assignedTo: PersonRelateToOneForUpdateInput
+  tags: TagRelateToManyForUpdateInput
+  finishBy: String
+}
+
+input TaskCreateInput {
+  label: String
+  priority: TaskPriorityType
+  isComplete: Boolean
+  assignedTo: PersonRelateToOneForCreateInput
+  tags: TagRelateToManyForCreateInput
+  finishBy: String
+}
+
+input PersonRelateToOneForUpdateInput {
+  create: PersonCreateInput
+  connect: PersonWhereUniqueInput
+  disconnect: Boolean
+}
+
+input PersonRelateToOneForCreateInput {
+  create: PersonCreateInput
+  connect: PersonWhereUniqueInput
+}
+
+input TagRelateToManyForUpdateInput {
+  disconnect: [TagWhereUniqueInput!]
+  set: [TagWhereUniqueInput!]
+  create: [TagCreateInput!]
+  connect: [TagWhereUniqueInput!]
+}
+
+input TagRelateToManyForCreateInput {
+  create: [TagCreateInput!]
+  connect: [TagWhereUniqueInput!]
+}
+```
+
+## Upgrade Checklist
+
+While there are a lot of changes to this API, if you approach the upgrade systematically you will be able to have a smooth upgrade.
+
+- Before upgrading, make sure you're not using any of the deprecated features, or the `search` argument to filters.
+- Update your top level queries. Make sure that you've renamed `Task` to `task` and `allTasks` to `tasks` for each query.
+- Update your filters. Each of the old filters will have a corresponding new filter, which can be directly replaced.
+- Update your mutation arguments to match the new input types. Make sure you've replaced `{ id: "..."}` with `{where: { id: "..."} }` in your `update` and `delete` operations.
+- Update your relationship inputs to `create` and `update` operations. Make sure you've replaced usage of `{ disconnectAll: true }` with `{ set: [] }` in to-many relationships, and have used `{ disconnect: true }` rather than `{ disconnect: { id: "..."} }` in to-one relationships.
+
+!> Make sure you make the corresponding changes to filters and input arguments when using the [List Items API](docs/apis/list-items).
+
+!> If you get stuck, or want to discuss these changes, reach out to us in the [Keystone community slack](https://community.keystonejs.com/).
```

Comment:
```suggestion
## Upgrade Checklist

While there are a lot of changes to this API, if you approach the upgrade process systematically your experience should be pretty smooth. If you get stuck or have questions, reach out to us in the [Keystone community slack](https://community.keystonejs.com/) to get the help you need.

?> Before you begin: check that your project doesn't rely on any of the features we've marked as deprecated in this document, or the `search` argument to filters. If you do, apply the recommended substitute.

1. Update top level queries. Be sure to rename `Task` to `task` and `allTasks` to `tasks` for all your queries.
2. Update filters. Find and replace all the old Keystone filters with their new equivalent.
3. Update mutation arguments to match the new input types. Make sure you replace `{ id: "..."}` with `{where: { id: "..."} }` in your `update` and `delete` operations.
4. Update relationship inputs to `create` and `update` operations. Ensuring you've replaced usage of `{ disconnectAll: true }` with `{ set: [] }` in to-many relationships, and have used `{ disconnect: true }` rather than `{ disconnect: { id: "..."} }` in to-one relationships.

!> Finally, make sure you apply corresponding changes to filters and input arguments when using the [List Items API](docs/apis/list-items).

---

That's everything! While we acknowledge that API changes are an inconvenience, we believe the time spent navigating these upgrades will be offset many times over by a more fun and productive developer experience going forward.

```